File: tp/61319 - Carabajal, José Gabriel/tp6/backend/models/compras.py
from typing import Optional
from sqlmodel import SQLModel, Field, Relationship
from datetime import datetime

class Compra(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    usuario_id: int
    fecha: datetime
    direccion: str
    tarjeta: str   
    subtotal: float
    iva: float
    envio: float
    total: float

class CompraItem(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    compra_id: int = Field(foreign_key="compra.id")
    producto_id: int
    nombre: str
    precio_unitario: float
    cantidad: int
    categoria: Optional[str] = None   
    iva: float = 0.0                  

File: tp/61271 - Donelli, Gerardo Exequíel/tp6/frontend/app/services/compras.ts
/**
 * Servicio de compras para comunicarse con el backend
 */

const API_URL = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000';

export interface ItemCompra {
  id: number;
  producto_id: number;
  cantidad: number;
  nombre: string;
  precio_unitario: number;
  subtotal: number;
  imagen: string;
}

export interface CompraResumen {
  id: number;
  fecha: string;
  total: number;
  envio: number;
  direccion: string;
  cantidad_productos: number;
}

export interface CompraDetalle {
  id: number;
  fecha: string;
  direccion: string;
  tarjeta: string;
  total: number;
  subtotal: number;
  iva: number;
  envio: number;
  items: ItemCompra[];
}

export interface HistorialResponse {
  compras: CompraResumen[];
  total_compras: number;
  mensaje?: string;
}

/**
 * Obtener historial de compras del usuario
 */
export async function obtenerHistorialCompras(token: string): Promise<HistorialResponse> {
  const response = await fetch(`${API_URL}/compras`, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${token}`,
    },
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.detail || 'Error al obtener historial de compras');
  }

  return response.json();
}

/**
 * Obtener detalle de una compra específica
 */
export async function obtenerDetalleCompra(
  token: string,
  compraId: number
): Promise<CompraDetalle> {
  const response = await fetch(`${API_URL}/compras/${compraId}`, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${token}`,
    },
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.detail || 'Error al obtener detalle de compra');
  }

  return response.json();
}
